src/tls/openssl.c

```diff
@@ -529,9 +529,15 @@ static int tls_net_handshake(struct flb_tls *tls,
             pthread_mutex_unlock(&ctx->mutex);
             return -1;
         }
-
-        if (vhost != NULL) {
-            SSL_set_tlsext_host_name(session->ssl, vhost);
+        if (tls->vhost != NULL) {
```

Comment:
Add a new field to `struct tls_session` to save the value that `tls_context_create` gets in the `verify` argument because this conditional block depends on that, otherwise you will override the default behavior.

Reply:
@leonardo-albertovich Field was already present in tls. Hence did not add any new field